Marlin/Marlin_main.cpp

```diff
@@ -5414,6 +5388,188 @@ inline void gcode_M226() {
   }
 #endif
 
+/*
+* M242 - General I2C Message Interface
+*   A - 4 - 127 7-bit decimal device address
+*   P - 0 - 255 Process ID (See I2C Commands in Configuration_adv.h)
+*   D - 0 - 255 Data to write
+*   E - 0 - 255 EEPROM address, if applicable
+*/
+inline void gcode_M242() {
+  int verbose_level = code_seen('V') || code_seen('v') ? code_value_short() : 0;
+  if (verbose_level < 0 || verbose_level > 4) {
+    SERIAL_ECHOLNPGM("?(V)erbose Level is implausible (0-4).");
+    return;
+  }
+
+  // Used to see if we've been given arguments, and to warn you through the
+  // serial port if they're not seen.
+  bool hasA, hasP, hasD, hasE;
+  uint8_t i2c_address;
+  uint8_t i2c_process_id;
+  uint8_t i2c_eeprom_address;
+  uint8_t i2c_data;
+  
+  // Desired address for peripheral device
+  if (hasA = code_seen('A')) {
+    i2c_address = (uint8_t)code_value();
+  }
+  // Desired process command given
+  if (hasP = code_seen('P')) {
+    i2c_process_id = (uint8_t)code_value();
+  }        
+  // Desired data given
+  if (hasD = code_seen('D')) {
+    i2c_data = (uint8_t)code_value();
+  }
+
+  // EEPROM address, if applicable
+  if (hasE = code_seen('E')) {
+    i2c_eeprom_address = (uint8_t)code_value();
+  }
+
+  if (!hasA){
+    SERIAL_ECHOLNPGM("No peripheral address given");
+  }
+  if (!hasP){
+    SERIAL_ECHOLNPGM("No process command given");
+  }
+  if (!hasD){
+    SERIAL_ECHOLNPGM("No data given");
+  }
+  if (!hasE){
+    SERIAL_ECHOLNPGM("No eeprom address given");
```

Comment:
So I added returns in the other inputs, if they're not given the function will return without issuing the command. If no eeprom address is given, a warning will be given but the function won't just return. This way, one doesn't need to be given to send a function but you'll be made aware you didn't enter one in case it was necessary.